Ho workflow updates (#86)
hassanobeid1994/tr_b_causal_2020: references/organization/Project_story_outline.md

---
jupyter:
  jupytext:
    formats: ipynb,md
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.2'
      jupytext_version: 1.3.4
  kernelspec:
    display_name: Python 3
    language: python
    name: python3
---

# <ins>Story</ins>

### Part 1: Selection on observables

> We show that an outcome model, without a data generation model, is not enough when it comes to making causal inferences. Inferences from an outcome model alone assume a very specific data-generation process (namely, one where all the covariates are independent), and it is important to make that clear for transparency. 

Reviewer: ```suggestion
We show that an outcome model, without a proper DAG outlining the data generating process, is not enough when it comes to making causal inferences. Inferences from an outcome model alone assume a data-generation process  where all covariates are independent and might not reflect the natural data generation process.
```

Reviewer: Also, I was a little confused by this part of your second sentence: "and it is important to make that clear for transparency." 

I have it removed just because I didn't know how to blend it in with my suggestions. Feel free to insert it back in a way that flows.

> We show that an outcome model, without a data generation model, is not enough when it comes to making causal inferences. Inferences from an outcome model alone assume a very specific data-generation process (namely, one where all the covariates are independent), and it is important to make that clear for transparency. 

Reviewer 2: ```suggestion
We show that an outcome model, without a specified data generation model, is not sufficient to make credible causal inferences. Inferences from an outcome model alone assume a very specific data-generation process (namely, one where all covariates generated independently), and it is important to make this clear for transparency. 
```

> To do that, we simulate a mode-choice dataset from a known outcome (MNL) model as well as an assumed causal graph. We then re-estimate a logit model on the simulated dataset, and try to estimate the causal effect of perturbing travel distance on the mode share of auto, using two ways:

Reviewer: ```suggestion
To do that, we simulate a mode-choice dataset based on: 1) data from the paper by Brathwaite & Walker (2018), 2) a known Multinomial Logit (MNL) outcome model, and 3) different assumed causal graphs. We then re-estimate a MNL model on the simulated dataset, and try to estimate the average causal effects of perturbing travel distance on the mode share of auto based modes under the different assumed causal graphs. More specifically:
```

> To do that, we simulate a mode-choice dataset from a known outcome (MNL) model as well as an assumed causal graph. We then re-estimate a logit model on the simulated dataset, and try to estimate the causal effect of perturbing travel distance on the mode share of auto, using two ways:

Reviewer 2: ```suggestion
To make our point, we simulate a mode-choice dataset from an assumed causal graph and a known outcome model--a multinomial logit (MNL) model. We then re-estimate the MNL model on the simulated dataset, and we estimate the causal effect of perturbing travel distance on the mode share of auto, using two ways:
```

>  - The first is when the change in travel distance doesn't affect any of the outcome variables.

Reviewer: ```suggestion
 - Under a causal graph where all covariates are independent, a perturbation of travel distance doesn't affect any of the remaining covariates specified in the model.
```

>  - The first is when the change in travel distance doesn't affect any of the outcome variables.
>  - The second is when the change in travel distance leads to a change in its descendants (namely like travel cost and time). Here, we're assuming that we somehow were able to know the causal graph as well as the parametric relationships between its nodes. In reality however, one needs to recover the causal graph by iterating through reasonable alternatives and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, of course using appropriate tests of statistical goodness-of-fit. 

Reviewer: ```suggestion
 - Under a causal graph where all covariates are independent, a perturbation of travel distance doesn't affect any of the remaining covariates specified in the model.
 - Under a causal graph showing causal relationships between covariates, a perturbation of travel distance leads to a change in its descendants (namely like travel cost and time). Here, we're assuming that we somehow know the true causal graph (and therefore the data generation process) as well as the parametric relationships between its nodes. In reality, however, one needs to recover the causal graph by iterating through reasonable alternative causal graphs and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a well-tested causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, all while using appropriate tests of statistical goodness-of-fit. 
```

>  - The second is when the change in travel distance leads to a change in its descendants (namely like travel cost and time). Here, we're assuming that we somehow were able to know the causal graph as well as the parametric relationships between its nodes. In reality however, one needs to recover the causal graph by iterating through reasonable alternatives and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, of course using appropriate tests of statistical goodness-of-fit. 

Reviewer: ```suggestion
- Under a causal graph showing causal relationships between covariates, a perturbation of travel distance leads to a change in its descendants (namely like travel cost and time). Here, we're assuming that we somehow know the true causal graph (and therefore the data generation process) as well as the parametric relationships between its nodes. In reality, however, one needs to recover the causal graph by iterating through reasonable alternative causal graphs and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a well-tested causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, all while using appropriate tests of statistical goodness-of-fit.
```

>  - The second is when the change in travel distance leads to a change in its descendants (namely like travel cost and time). Here, we're assuming that we somehow were able to know the causal graph as well as the parametric relationships between its nodes. In reality however, one needs to recover the causal graph by iterating through reasonable alternatives and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, of course using appropriate tests of statistical goodness-of-fit. 

Reviewer 2: ```suggestion
 - The second is when the change in travel distance leads to a change in its descendants (namely travel cost and time). Here, we're assuming that we know the causal graph as well as the parametric relationships between its nodes. In reality however, one needs to recover the causal graph by iterating through reasonable alternatives and attempting to falsify them using the conditional independence tests outlined [here: reference to Timothy cit notebooks]. Once a causal graph is obtained, the modeler has to then model the dependence between the variables using any parametric or non-parametric specifications that she deems appropriate, of course using appropriate tests of statistical goodness-of-fit. 
```

> We run the process above for N simulations and obtain a distribution of the causal effect of interest. The purpose of this exercise is to show that talking about causal effects without explicitely having a model of the data generation process leads to significant bias in the results. 

Reviewer: ```suggestion
We run the process above for N simulations and obtain a distribution of the average causal effect of interest. The purpose of this exercise is to show that talking about causal effects without explicitly specifying a model of the data generation process leads to biased results. 
```

Reviewer: I only removed significant because as we saw last week, we are yet to quantify what is considered significant bias.

Possible extensions:
So what we've done now is show that a causal effect estimated "naively" is biased, but we haven't shown how to construct the causal graph, or estimate the relationships between the nodes, which the modeler has to do. I'm thinking we can present the problem using three levels of hierarchy for contributions:
 - We don't know the causal graph. 
 - We know the causal graph, but we don't know the equations on the edges. 
 - We know the causal graph, and we also know the equations on the edges. 

> Our work so far has focused on level 3. Is there value of a demonstration for level two? Or should we just outline the steps?

Reviewer: I think we should. Not sure about the feasibility within our current timeline of June 1st?

Reviewer 2: I think we should simply outline the fact that we need to (i) generate causal graphs that are not falsified by our data, and (ii) estimate the distributions of each node conditioned on its parents.

> ### Part 2: Selection on observables

Reviewer: ```suggestion
### Part 2: Dealing with confounding
```

Reviewer 2: ```suggestion
### Part 2: Dealing with unobserved confounding
```
